autonomy/cli/helpers/chain.py

```diff
@@ -0,0 +1,195 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2022 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""On-chain interaction helpers."""
+import json
+from collections import OrderedDict
+from pathlib import Path
+from typing import Any, Dict, List, Optional, cast
+
+import click
+from aea.configurations.constants import DEFAULT_README_FILE
+from aea.configurations.data_types import PackageType, PublicId
+from aea.configurations.loader import load_configuration_object
+from aea.crypto.base import Crypto, LedgerApi
+from aea.helpers.cid import CID
+from aea.helpers.ipfs.base import IPFSHashOnly
+from aea_cli_ipfs.ipfs_utils import IPFSTool
+from aea_ledger_ethereum.ethereum import EthereumApi
+
+from autonomy.chain.base import ComponentRegistry, RegistryManager
+from autonomy.chain.config import COMPONENT_REGISTRY_LOCAL, REGISTRIES_MANAGER_LOCAL
+from autonomy.configurations.base import PACKAGE_TYPE_TO_CONFIG_CLASS
+
+
+BASE16_HASH_PREFIX = "f01701220"
+CONFIG_HASH_STRING_PREFIX = "0x"
+UNIT_HASH_PREFIX = CONFIG_HASH_STRING_PREFIX + "{metadata_hash}"
+DEFAULT_NFT_IMAGE_HASH = "bafybeiggnad44tftcrenycru2qtyqnripfzitv5yume4szbkl33vfd4abm"
+
+ContractInterfaceType = Any
+
+
+def serialize_metadata(
+    package_hash: str,
+    public_id: PublicId,
+    description: str,
+    nft_image_hash: str = DEFAULT_NFT_IMAGE_HASH,
+) -> str:
+    """Serialize metadata."""
+    metadata = OrderedDict(
+        {
+            "name": str(public_id),
```

Comment:
Public id without hash

Reply:
And without version imo

Reply:
I agree with the hash part but I don't understand why we don't want to include the version part

Reply:
Addressed [cd6c044](https://github.com/valory-xyz/open-autonomy/pull/1619/commits/cd6c0442379700abd7137d033e0d5e6a475253e0)